docs/decisions/0014-single-taxonomy-view-api.rst

```diff
@@ -0,0 +1,156 @@
+13. Single taxonomy view API
+=====================================
+
+
+Context
+--------
+
+This view returns tags of a closed taxonomy (for MVP has not been implemented yet
+for open taxonomies). It is necessary to make a decision about what structure the tags are going 
+to have, how the pagination is going to work and how will the search for tags be implemented.
+It was taken into account that a taxonomy can have a large number of tags and are mostly represented as trees.
+
+**Branch:** Is the representation of a root Tag and all its children up to the leaves.
+
+For the decisions, the following use cases were taken into account:
+
+- List of root tags that can be expanded to show children tags.
+    - This list can be sorted alphabetically: A-Z (default) and Z-A
+- The user can expand all root tags.
+- The user can search for tags.
```

Comment:
We need to clarify in this ADR how this search is going to work, and how it interacts with pagination.

My suggestion is something like this:
1. On the backend when serving this API, if there is a search term like "cat", filter all the tags to only those that contain the string "cat" as well as their ancestor tags.
2. If the total number that matched is less than ~200, send them all to the frontend in a JSON tree structure.
3. If the total number that matched is more than that, send only [the first page of] root tags to the frontend. Each root tag can be expanded by another API call, and/or additional pages of root tags can be loaded.

But let's discuss this with product and UX team before making too many changes here.

Reply:
@bradenmacdonald I have updated the document with what was discussed. 

I've been thinking, if we are going to return complete taxonomies that have less than 1000 tags, we can do a search in the frontend with these taxonomies. I have not added it to the document, because I think that many conditionals are already being added and it will complicate the implementation. What do you think?

Reply:
Yes, I think so too. Although I also don't want to complicate things too much by having two different implementations of search, based on the taxonomy size. Maybe there's some way we can make the "< 1,000 so load all and search via frontend" just a subset / special case of the general version. Something like "if all tags are loaded into the frontend, do a local filter, else do a server-side search and refresh the tags shown" would work in both cases.